### DISCUSSION TITLE
IDE integrated remote debugging options (vscode) #76

### DISCUSSION THREAD

**Discussion by kowh-ai** [ckan/ckan-docker#122] [Q&A]
Following on from https://github.com/okfn/docker-ckan/issues/76

**Comment by AyanIanniello** [comment]
I think the bash command that worked to launch debugpy for ckan 2.9 needs to be adapted for ckan 2.10 or maybe other part of ckan. 
Here is what I did: 

(1) exposed a second port 5678 in the DockerFile:
```
services:
   ckan-dev:
	[...]
	ports:
      	- "0.0.0.0:${CKAN_PORT_HOST}:${CKAN_PORT}"
      	- "0.0.0.0:5678:5678"  # added line
```

(2) made a copy of the file _ckan-docker-base/ckan-2.10/dev/setup/start_ckan_development.sh_  in _ckan-docker/ckan/setup/start_ckan_development.sh_ and added the lines below to the _ckan/Dockerfile.dev_ which will overwrite the _start_ckan_development.sh_ it pulls from ckan base 2.10.3

```
COPY ./setup/start_ckan_development.sh ${APP_DIR}/start_ckan_development.sh
RUN chmod +x ${APP_DIR}/start_ckan_development.sh
```

(3) Updated the ckan-docker/ckan/setup/start_ckan_development.sh with:

```
pip install debugpy
sudo ckan -EH /usr/bin/python -m debugpy --log-to-stderr --wait-for-client --listen 0.0.0.0:5678 /usr/bin/ckan serve --reload $CKAN_INI
```
Build and run the docker containers

```
docker compose -f docker-compose.dev.yml build
docker compose -f docker-compose.dev.yml up
```
I hit various error from here and tried slight changes (among other things replaced sudo with su command, tried ckan -c run  $CKAN_INI instead of serve --reload). but I just don't understand enough ckan to adapt this line.

Will watch this space and hope someone will share how they made it work for them 🙂

**Reply by wardi** [reply]
@AyanIanniello I've added this feature in https://github.com/ckan/ckan-docker-base/pull/50 and documented it in https://github.com/ckan/ckan-docker/pull/139/commits/275a9374a7f152b79babe6b7f27709d10fac1d8a
please give it a try and let me know if it's working for you.
